meta-mbl-distro/scripts/lib/mbl/update/appsimage.py

```diff
@@ -46,12 +42,69 @@ def generate_testinfo(self):
     @property
     def image_type(self):
         """Implement method from PayloadImage ABC."""
-        return "{}v3".format(MBL_APPS_ID)
+        return "{}v{}".format(MBL_APPS_ID, self._image_format_version)
+
+
+class TarXzAppsImage(_AppsImageBase):
+    """
+    Class for creating .tar.xz image files containing apps (ipk files).
+
+    With this image format, all ipks are bundled together in a .tar.xz
+    archive before being added to the payload archive.
+    """
+
+    def __init__(self, app_paths):
+        """
+        Create a TarXzAppsImage object.
+
+        Args:
+        * app_paths list<str>: list of paths to the ipk files to add to the
+          payload.
+
+        """
+        super().__init__(app_paths, image_format_version=3)
```

Comment:
Should we have some kind of global data structure holding the format versions? We could have a single `namedtuple` which contains the format version for each image. I think it's starting to become a bit of a maintenance headache to have to look through all the class definitions and change the versions.

Reply:
https://github.com/ARMmbed/meta-mbl/pull/781/commits/d0575f20da4a268148336cd94f5dbbb876d52437